### PR TITLE
Make provisioning faster

### PR DESCRIPTION
**Pull request by andyliuliming**
1. make some tasks in the agent run in parallel to speed up the provisioning.

2. adjust the interval of the checking period to speed up the provisioning.

### CODE REVIEW

File: azurelinuxagent/common/protocol/util.py [hunk 1]
```diff
@@ -144,50 +160,75 @@ def _set_wireserver_endpoint(self, endpoint):
...
-                self.dhcp_handler.run()
-            except DhcpError as e:
-                raise ProtocolError(ustr(e))
+        """
+        detect the wire protocol.
+        """
+        self.wire_lock.acquire()
+        try:
+            if self.wire_protocol != None:
+                return self.wire_protocol
```
Comment (jasonzio) [inline azurelinuxagent/common/protocol/util.py]: Use `if x is not None`; do not use `if x != None`. PEP-8 compliance, proper use of python, etc.

**Comment by andyliuliming** [comment]
@jasonzio tested only on the ubuntu xenial and our stemcell(CF) for now. I have dungen crawler environment, could you please tell me which scenario to run for the full test?

File: azurelinuxagent/common/logger.py [hunk 2]
```diff
@@ -77,11 +78,12 @@ def log(self, level, msg_format, *args):
...
         time = datetime.now().strftime(u'%Y/%m/%d %H:%M:%S.%f')
         level_str = LogLevel.STRINGS[level]
+        thread_name = threading.currentThread().name
         if self.prefix is not None:
-            log_item = u"{0} {1} {2} {3}\n".format(time, level_str, self.prefix,
+            log_item = u"{0} {1} {2} {3} {4}\n".format(time, thread_name, level_str, self.prefix,
                                                    msg)
         else:
-            log_item = u"{0} {1} {2}\n".format(time, level_str, msg)
+            log_item = u"{0} {1} {2} {3}\n".format(time, thread_name, level_str, msg)
```
Comment (hglkrijger) [inline azurelinuxagent/common/logger.py]: Fyi, these changes will break some of the dungeon crawler tests, which expect a specific log format

**Comment by andyliuliming** [comment]
@hglkrijger Thanks for the reviewing.
I will seperate the clean up with the "parallel" change, and send out the clean up pr first :)
and try to run the full bvt, and see the time metrics change.

**Comment by hglkrijger** [comment]
Closing this, since you opened a new PR

**Pull request closed without merging**